torch_np/_funcs_impl.py

```diff
@@ -1002,8 +1002,8 @@ def choose(
     n_dims = choices.ndim
```

Comment:
nit. now `n_dims` is not used, and you can directly use `choices.shape` below rather an aux variable.

Reply:
`torch.arange(dim).view((1,) * i + (dim,) + (1,) * (choices.ndim - i - 1)`, really? OK, done.